### PR TITLE
Use IndexedDB & Add Zoom Restriction Option

### PR DESCRIPTION
**Pull request by drg101**
Use IndexedDB, with the Dexie wrapper, to cache shapefiles. Advantages being:
- Doesn't use JS heap at all.
- The cached data will stay on the User's machine through reloads.

Also added ability to restrict zoom levels. Look at static attributes of `AutoQuery`. I've added a `DEV` boolean to map2.js where we set those zoom things to a less restricted amount.

### CODE REVIEW

**Comment by drg101** [comment]
Gonna add the aggregation thingy

File: src/iframe/js/frame-specific/map2.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 const MAPNUMBER = 2;
 const e = React.createElement;
 
-//const queryAlertText = document.getElementById('queryInfoText');
-
+const DEV = true;
```
Comment (Marc-9) [inline src/iframe/js/frame-specific/map2.js:4]: Wouldn't it be better to parse the domain url and check
Something like
const DEV = window.location.href.split("/")[1] === "dev"

**Review by Marc-9 — CHANGES_REQUESTED** [review]
Errors with Auto generated constraints
Electrical Substations- Uncaught TypeError: feature.properties is undefined
Power Plants/Hospitals- Uncaught Error: Invalid GeoJSON object.
Gas Pipelines- Uncaught TypeError: feature.properties is undefin in aperture util
Pretty much all the other also spam the console with invalid geojson object

File: src/iframe/js/frame-specific/map2.js (hunk 1 above)
Comment (drg101) [reply src/iframe/js/frame-specific/map2.js:4]: Yea, good idea

**Comment by drg101** [comment]
@Marc-9 addressed all of your issues

**Comment by drg101** [comment]
Going to add preloading

**Comment by drg101** [comment]
This thing is crazy fast now.

**Review by Marc-9 — APPROVED** [review]
Graph does not work, perhaps this will be fixed when merged
Also small css issue with the menu on screens <= 13 inches

**Comment by pierce-smith1** [comment]
Is zooming out supposed to be blocked in DEV mode? Because it doesn't seem to be.
Also, where is the preload status text showing up (is it supposed to show up at all?)

**Comment by pierce-smith1** [comment]
No idea why charting doesn't work because the code that deals with it hasn't changed, but we're going to have to deal with this problem regardless once the complete charting feature gets merged into dev. Boy, that's going to be fun.

**Comment by drg101** [comment]
@pierce-smith1 
> Is zooming out supposed to be blocked in DEV mode? Because it doesn't seem to be.
> Also, where is the preload status text showing up (is it supposed to show up at all?)
Zooming is unrestricted in DEV mode. The queries are honestly fast enough to zoom out a ton now anyways. 



> No idea why charting doesn't work because the code that deals with it hasn't changed, but we're going to have to deal with this problem regardless once the complete charting feature gets merged into dev. Boy, that's going to be fun.

All the data still goes through renderInfrastructure's renderGeoJson method, so that's weird. Which attributes are you pulling from the data, I've filtered the response data to only include what is going to be on the popup, which is likely your issue.

**Comment by drg101** [comment]
> Graph does not work, perhaps this will be fixed when merged
> Also small css issue with the menu on screens <= 13 inches

@honeybadgerofdoom Could you address the skinny menu issue in a separate PR?

**Pull request merged**